operator/internal/validation/lokistack.go

```diff
@@ -23,6 +26,7 @@ var _ admission.CustomValidator = &LokiStackValidator{}
 // LokiStackValidator implements a custom validator for LokiStack resources.
 type LokiStackValidator struct {
 	ExtendedValidator func(context.Context, *lokiv1.LokiStack) field.ErrorList
+	Client            client.Client
```

Comment:
I think you should move `Client` above `ExtendedValidator` as the extended validation is optional, but the client is not.